src/common/mfu_io.c

```diff
@@ -346,7 +737,52 @@ off_t mfu_lseek(const char* file, int fd, off_t pos, int whence)
     return rc;
 }
 
+off_t mfu_file_lseek(const char* file, mfu_file_t* mfu_file, off_t pos, int whence)
+{
+    if (mfu_file->type == POSIX) {
+        off_t rc = mfu_lseek(file, mfu_file->fd, pos, whence);
+        return rc;
+    } else if (mfu_file->type == DAOS) {
+        off_t rc = daos_lseek(file, mfu_file, pos, whence);
+        return rc;
+    } else {
+        MFU_ABORT(-1, "File type not known: %s errno=%d (%s)",
+                  file, errno, strerror(errno));
+    }
+}
+
 /* reliable read from file descriptor (retries, if necessary, until hard error) */
+ssize_t mfu_file_read(const char* file, void* buf, size_t size, mfu_file_t* mfu_file)
+{
+    if (mfu_file->type == POSIX) {
+        ssize_t got_size = mfu_read(file, mfu_file->fd, buf, size);
+        return got_size;
+    } else if (mfu_file->type == DAOS) {
+        ssize_t got_size = daos_read(file, buf, size, mfu_file);
+        return got_size;
+    } else {
+        MFU_ABORT(-1, "File type not known: %s errno=%d (%s)",
+                  file, errno, strerror(errno));
+    }
+}
+
+ssize_t daos_read(const char* file, void* buf, size_t size, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    daos_size_t got_size;
+    mfu_file->sgl->sg_iovs[0].iov_len = size;
+    int rc = dfs_read(mfu_file->dfs, mfu_file->obj, mfu_file->sgl, mfu_file->offset, &got_size, NULL); 
+    ssize_t size_read = (ssize_t)got_size;
```

Comment:
This line needed?

Reply:
@adammoody I suppose it can just return the got_size casted to the ssize_t type. I don't feel strongly about it either way.

Reply:
It's fine to return this instead.  I just didn't see ```size_read``` being used anywhere, but maybe I missed that.  Let's drop it if it's not actually used.

Reply:
@adammoody ok yeah I missed that. I removed that line.